src/MongoDB.EntityFrameworkCore/Serializers/EntitySerializer.cs

```diff
@@ -48,7 +53,41 @@ public TValue Deserialize(BsonDeserializationContext context, BsonDeserializatio
             => Deserialize(context, args);
 
         public void Serialize(BsonSerializationContext context, BsonSerializationArgs args, TValue value)
-            => throw new NotImplementedException();
+        {
+            if (value == null)
+            {
+                context.Writer.WriteNull();
+                return;
+            }
+
+            var storedKeyProperties = GetStoredKeyProperties();
+            if (storedKeyProperties.Any())
+            {
+                // Has key, throw and indicate user should compare by key
```

Comment:
Is there any better place for the code that is clearly related to entities comparison only? It's confusing why EntitySerializer trying to validate  comparison operation inside of Serialize method.

Reply:
This code would be removed once the comparison issue is solved next release.